IDE/Espressif/ESP-IDF/test/test_wolfssl.c

```diff
@@ -116,6 +165,473 @@ static void tskAes256_Test(void *pvParam)
+                        printf("X = %s Y = %s P = %s\n", mp_isodd(&x)==MP_YES?"ODD":"EVEN",
+                                                        mp_isodd(&y)==MP_YES?"ODD":"EVEN",
+                                                        mp_isodd(&p)==MP_YES?"ODD":"EVEN");
+                        TEST_FAIL_MESSAGE("mp_mulmod() failed");
+                    }
+                    elapsedTime2 = esp32elapsedTime();
+                }
+                if(exptmod) {
+                    /* Z = X^Y mod M */
+                    esp32TimerStart();
+                    if(ret == 0 && (ret = mp_exptmod(&x, &y, &p, &z3)) != MP_OKAY) {
+                        printf("(%d,%d) Xbits = %d, Ybits = %d Pbits = %d ret = %d\n",
+                                                    i , j, Xbits, Ybits, Pbits, ret);
+                        printf("X = %s Y = %s P = %s\n", mp_isodd(&x)==MP_YES?"ODD":"EVEN",
+                                                            mp_isodd(&y)==MP_YES?"ODD":"EVEN",
+                                                            mp_isodd(&p)==MP_YES?"ODD":"EVEN");
+                        TEST_FAIL_MESSAGE("mp_exptmod() failed");
+                    }
+                    elapsedTime3 = esp32elapsedTime()/1000;
+                }
+
+                printf("(%d,%d) Xbits = %d, Ybits = %d Pbits = %d",
+                                                        i , j, Xbits, Ybits, Pbits);
+                if(mul) {
+                    printf(" mul = %llu (us)", elapsedTime1);
+                }
+                if(mulmod) {
+                    printf(" mulmod = %llu (us)\n", elapsedTime2);
+                }
+                if(exptmod) {
+                    printf(" exptmod = %llu (ms)\n", elapsedTime3);
+                }
+            }
+        }
+    }
+
+    mp_clear(&z1);
+    mp_clear(&z2);
+    mp_clear(&z3);
+    mp_clear(&x);
+    mp_clear(&p);
+    mp_clear(&y);
+
+    wc_FreeRng(&rng);
+
+    return ret;
+}
+
+/* Z = X * Y */
+int mp_unitest_mul(const char* strZ, const char* strX, const char* strY, int verbose)
+{
+    int ret = 0;
+    char* buf;
+    char* bufZ;
+    int radixX_size;
+    int radixZ_size;
+    int radixY_size;
+
+    mp_int z, x, y;
+
+     /* set init mp_int */
+    if((ret = mp_init_multi(&z, &x, &y, NULL, NULL, NULL)) != MP_OKAY) {
+        TEST_FAIL_MESSAGE("mp_init failed.\n");
+    }
+    /* set big-numbers into mp_inits */
+    if(ret == 0 && mp_read_radix(&x, strX, 16) != MP_OKAY) {
+        ESP_LOGE(TAG,"failed to set strX into mp_init");
+        return -1;
+    }
+    if(ret == 0 && mp_read_radix(&y, strY, 16) != MP_OKAY) {
+        ESP_LOGE(TAG,"failed to set strY into mp_init");
+        return -1;
+    }
+    /* Z = X * Y */
+    if(ret == 0 && (ret = mp_mul(&x, &y, &z)) != MP_OKAY) {
+        TEST_FAIL_MESSAGE("mp_mul() failed\n");
+    }
+
+    mp_radix_size(&z, 16, &radixZ_size);
+    bufZ = (char*)XMALLOC(radixZ_size + 1, NULL, DYNAMIC_TYPE_IN_BUFFER);
```

Comment:
Please use `DYNAMIC_TYPE_TMP_BUFFER` for this and other places below. Also the XFREE dynamic type needs to match. The DYNAMIC_TYPE_IN_BUFFER is used specifically for TLS input buffer processing.

Reply:
Changed to use DYANMIC_TYPE_TMP_BUFFER. Thank you for letting me know it.